runtime/sema/type.go

```diff
@@ -277,7 +277,7 @@ type LocatedType interface {
 type ParameterizedType interface {
 	Type
 	TypeParameters() []*TypeParameter
-	Instantiate(typeArguments []Type, report func(err error)) Type
+	Instantiate(typeArguments []Type, report func(err error), astRange *ast.Range) Type
```

Comment:
Currently, the full range of all type arguments is constructed and passed to the instantiation function.

We  should have the ranges of the individual type arguments already available, so maybe just pass in the AST type arguments to the function, and then the instantiation function can report errors for individual type arguments, instead just for the full range of all arguments.

Reply:
Done. Added that in the `Instantiate` function as well as in the constructor function.

Note that the type arguments are optional in the constructor function. So, I fallback to the invocation range in that case. 733cf4706